pkg/api/validate/openshiftcluster_validatedynamic.go

```diff
@@ -110,33 +133,27 @@ func (dv *openShiftClusterDynamicValidator) Dynamic(ctx context.Context) error {
 		return err
 	}
 
-	err = dv.validateVnetPermissions(ctx, spAuthorizer, dv.spPermissions, vnetID, &vnetr, api.CloudErrorCodeInvalidServicePrincipalPermissions, "provided service principal")
-	if err != nil {
-		return err
-	}
-
-	err = dv.validateVnetPermissions(ctx, dv.fpAuthorizer, dv.fpPermissions, vnetID, &vnetr, api.CloudErrorCodeInvalidResourceProviderPermissions, "resource provider")
+	err = dv.ValidateVnetPermissionsSP(ctx, vnetID, &vnetr)
 	if err != nil {
 		return err
 	}
 
-	// Get after validating permissions
-	vnet, err := dv.spVirtualNetworks.Get(ctx, vnetr.ResourceGroup, vnetr.ResourceName, "")
+	err = dv.ValidateVnetPermissionsRP(ctx, vnetID, &vnetr)
 	if err != nil {
 		return err
 	}
 
-	err = dv.validateRouteTablePermissions(ctx, spAuthorizer, dv.spPermissions, &vnet, api.CloudErrorCodeInvalidServicePrincipalPermissions, "provided service principal")
+	err = dv.ValidateRouteTablePermissionsSP(ctx, &vnetr)
 	if err != nil {
 		return err
 	}
 
-	err = dv.validateRouteTablePermissions(ctx, dv.fpAuthorizer, dv.fpPermissions, &vnet, api.CloudErrorCodeInvalidResourceProviderPermissions, "resource provider")
+	err = dv.ValidateRouteTablePermissionsRP(ctx, &vnetr)
 	if err != nil {
 		return err
 	}
 
-	err = dv.validateVnet(ctx, &vnet)
+	err = dv.ValidateVnet(ctx, &vnetr)
```

Comment:
in this way the user of this package does not need to retrieve the vnet which will require setting up the required authorizers